src/coreComponents/mesh/MeshManager.cpp

```diff
@@ -61,21 +62,32 @@ void MeshManager::generateMeshes( DomainPartition & domain )
   forSubGroups< MeshGeneratorBase >( [&]( MeshGeneratorBase & meshGen )
   {
     MeshBody & meshBody = domain.getMeshBodies().registerGroup< MeshBody >( meshGen.getName() );
-    meshGen.generateMesh(meshBody); meshBody.createMeshLevel( 0 );
+    meshBody.createMeshLevel( 0 );
+    MeshLevel & meshLevel = meshBody.getBaseDiscretization();
     CellBlockManager & cellBlockManager = meshBody.registerGroup< CellBlockManager >( keys::cellManager );
 
+    meshGen.generateMesh( cellBlockManager );
+    meshBody.setGlobalLengthScale( cellBlockManager.getGlobalLength() );
+    ElementRegionManager & elemManager = meshLevel.getElemManager();
+    for ( WellBlock wellBlock : cellBlockManager.getWellBlocks() ) {
+      WellElementRegion &
+        wellRegion = elemManager.getGroup( ElementRegionManager::groupKeyStruct::elementRegionsGroup() ).
+        getGroup< WellElementRegion >( wellBlock.getWellRegionName() );
+      wellRegion.setWellGeneratorName( wellBlock.getName() );
+      wellRegion.setWellControlsName( wellBlock.getWellControlsName() );
+    }
```

Comment:
It's really cool to have this on this side of the mirror.
But I'm not sure this is the final place where this should be.
We can figure this out later.